### ISSUE TITLE
Arch Linux Install / Flatpak

### ISSUE THREAD

**Issue by HidYn** [nvdweem/PCPanel#16]
Hi Guys,

Thank you so much for developing this. I had actually put off ordering a PCPanel as I use Linux but thanks to your hard work I've just ordered one.

I plan to use it on Arch Linux. Would it be possible to have some instructions to install on that or maybe a Flatpak install?

Thanks

**Comment by nvdweem** [comment]
I've tried Linux a few times but haven't actually switched away from using Windows, so apart from having made the software compatible I'm far from a Linux expert so I'm hoping that the Linux community is knowledgeable enough to make it work :). 

There are some instructions on how to install the deb file which has been tested on Ubuntu and another user has tried it on Pop! but that distribution doesn't use pulse_audio by default so it won't allow volume control. That's also where my knowledge kind of stops, I don't know if it's possible to make a distribution use pulse_audio or not or how to control volume on systems that use something else.

I've not used Arch so I'm not sure if that allows installing deb files or if it uses pulse_audio. The software is written in Java so it should be able to run on any Java supporting OS.
I'll try to make some time to install an Arch VM and try some things out.

**Comment by frantzeskidis** [comment]
What a coincidence.. Yesterday tried installing on debian sid but gnome 43 is using wayland with pipewire and only the ui worked. Would love to a see flatpak or an rpm (for fedora) that works with pipewire.

**Comment by nvdweem** [comment]
It seems that the package method that is used (jpackage) isn't compatible with Flatpack so I added the jar file as an artifact and some instructions on how to run it.

The software used `pactl` for controlling the volume, but also `pacmd` for getting the devices. The latter isn't in the default PipeWire compatibility tools so I changed it's usage to also use `pactl`. This means that the software should now work when PipeWire is being used.

This was tested with the EndeavourOS build that was given.

**Comment by kastoreion** [comment]
It works on Debian-Sid flawlessly! Thank you!

If you are using gnome u need a tray extension otherwise the app is not launching.

**Comment by nvdweem** [comment]
The tray extension should not be needed anymore, but without a tray icon it's currently not possible to nicely stop the software (but why would you want to ;)).
Showing the main panel can be done by just running the software again, it should only keep 1 instance open.

**Comment by nvdweem** [comment]
`Cannot run program "xdotool": error=2, No such file or directory` should indicate that xdotool isn't installed. iirc, that tool is used to determine which program has focus for focus volume. I'd expect that any knob/slider that has the 'Focus Volume' option set shouldn't work unless xdotool is installed.

**Comment by kastoreion** [comment]
Definitely, can confirm on Fedora 37 with Wayland (NVIDIA proprietary 525.53) and using the .jar file for instance focus volume does not work, also visual glitches when moving the window, new window might go black, etc.

**Comment by kastoreion** [comment]
https://github.com/ReimuNotMoe/ydotool

**Comment by nvdweem** [comment]
I'll try and see if I can implement the ydotool soon. iirc, the distro I used last also has a Wayland option so I should be able to test the changes.

**Comment by nvdweem** [comment]
I'm looking into the `ydotool` but it seems that it only supports a part of `xdotool` that isn't useful for this usecase:
> Currently implemented command(s):
>
> type - Type a string
> key - Press keys
> mousemove - Move mouse pointer to absolute position
> click - Click on mouse buttons

For the focus volume, we need an alternative for the `getactivewindow` and `getwindowpid` options of `xdotool`. There is an issue that asks about this specifically but it was answered to be impossible: https://github.com/ReimuNotMoe/ydotool/issues/56

The same seems to apply to the `wtype` suggestion: `xdotool type for wayland`, we don't need the `type` part of `xdotool`.

From what I can tell, there doesn't seem to be a way to get the processname or pid for the window that has focus in Wayland. There are some mentions about it being possible on `Sway` but I have no idea if that is something that is commonly used. My distro with GNOME Wayland doesn't seem to use it.

**Comment by ethan-neidhart37** [comment]
Hello! 

Sorry to resurrect an old thread like this but I figured it would be better than opening a new issue when this one already exists.

Would you be open to publishing this to the AUR? If flatpak isn't an option then the AUR is probably the simplest path to a smoother installation process for arch based distros.

I have no experience with this kind of thing but I'd be happy to help out and learn along the way. From a very cursory Google search about how publishing to the AUR works, I'm hopeful that it would be a fairly simple process!

**Comment by nvdweem** [comment]
I'm going to close this issue. I think most of the mentioned issues should have been resolved by now. Remaining flatpak issues are being picked up in #88
